.github/workflows/golang.yml

```diff
@@ -1,6 +1,13 @@
 name: golang
 on:
   push:
+    branches:
+      - develop
```

Comment:
This allows other branches to use the workflows because of the `pull_request` section below, right?

Reply:
Yes, and we wont have duplicated pipelines under a PR because of this, and we wont run these for every commit on every branch just when the given branch is near finished(PR submitted).